### PR TITLE
fix(ci): embed v0.9.20 runtime, update test suite, and ci workflows

### PR DESCRIPTION
**Pull request by timwu20**
## Changes

- embeds 0.9.20 runtime
- updated unit tests that require old runtimes (gssmr-v3substate, dev-v3substrate)
- updated all ci workflows
- updated integration tests

## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Issues

completes #2540 

## Primary Reviewer

@jimjbrettj

### CODE REVIEW

**Comment by danforbes** [comment]
Linked to #2540

File: chain/dev/config.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ digest = ""
 genesis = "./chain/dev/genesis-spec.json"
 
 [account]
-key = "alice"
+key = ""
```
Comment (qdm12) [inline chain/dev/config.toml:21]: Are we sure we don't want a key anymore?

File: chain/westend-dev/defaults.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
+)
+
+var (
+	// GlobalConfig
+
+	// DefaultName Default node name
+	DefaultName = string("Gossamer")
+	// DefaultID Default chain ID
+	DefaultID = string("gssmr")
```
Comment (qdm12) [inline chain/westend-dev/defaults.go:20]: should this be westend?

File: chain/westend-dev/defaults.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	// DefaultConfig Default toml configuration path
+	DefaultConfig = string("./chain/gssmr/config.toml")
```
Comment (qdm12) [inline chain/westend-dev/defaults.go:22]: should this be

```suggestion
	DefaultConfig = string("./chain/westend-dev/config.toml")
```

File: chain/westend-dev/defaults.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	// DefaultRetainBlocks is the default retained blocks
+	DefaultRetainBlocks = int64(512)
+
+	// DefaultTelemetryURLs is the default URL of the telemetry server to connect to.
+	DefaultTelemetryURLs []genesis.TelemetryEndpoint
+
+	// InitConfig
+
+	// DefaultGenesis is the default genesis configuration path
+	DefaultGenesis = string("./chain/gssmr/genesis-spec.json")
```
Comment (qdm12) [inline chain/westend-dev/defaults.go:43]: should this be

```suggestion
	DefaultGenesis = string("./chain/westend-dev/westend-dev-spec.json")
```

File: cmd/gossamer/import_runtime.go [hunk 5]
```diff
@@ -29,7 +29,7 @@ func createGenesisWithRuntime(fp string) (string, error) {
 		return "", err
 	}
 
-	genesis.Genesis.Runtime["System"]["code"] = fmt.Sprintf("0x%x", runtime)
+	genesis.Genesis.Runtime["system"]["code"] = fmt.Sprintf("0x%x", runtime)
```
Comment (qdm12) [inline cmd/gossamer/import_runtime.go:32]: Out of curiosity, was this right before? Or did this change with newer runtimes?

File: cmd/gossamer/main_test.go [hunk 6]
```diff
@@ -268,7 +268,7 @@ func TestBuildSpecCommandWithOutput(t *testing.T) {
 	buildSpecCommand := runTestGossamer(t,
 		"build-spec",
 		"--raw",
-		"--genesis-spec", utils.GetGssmrGenesisPathTest(t),
+		"--genesis-spec", utils.GetGssmrGenesisRawPathTest(t),
```
Comment (qdm12) [inline cmd/gossamer/main_test.go:271]: nit why the change? 🤔

File: dot/network/gossip_test.go [hunk 7]
```diff
@@ -84,7 +84,9 @@ func TestGossip(t *testing.T) {
 		Digest: types.NewDigest(),
 	}
 
-	_, err = nodeA.host.send(addrInfoB.ID, "", announceMessage)
+	time.Sleep(TestMessageTimeout)
+
+	_, err = nodeA.host.send(addrInfoB.ID, "/gossamer/test/0/block-announces/1", announceMessage)
```
Comment (qdm12) [inline dot/network/gossip_test.go:89]: Why do we need a sleep now?

File: dot/rpc/modules/childstate_test.go [hunk 8]
```diff
@@ -42,47 +42,14 @@ func createTestTrieState(t *testing.T) (*trie.Trie, common.Hash) {
...
-		"0xe2e62dd81c48a88f73b6f6463555fd8eba7fb8745735dc3be2a2c61a72c39e78"}
-
 	tr, sr := createTestTrieState(t)
 
+	expKeys := tr.GetKeysWithPrefix([]byte{})
+	expHexKeys := make([]string, len(expKeys))
+	for idx, k := range expKeys {
+		expHexKeys[idx] = common.BytesToHex(k)
+	}
+
```
Comment (qdm12) [inline dot/rpc/modules/childstate_test.go:52]: Noice 👍

File: go.mod [hunk 9]
```diff
@@ -176,17 +174,23 @@ require (
...
 	lukechampine.com/blake3 v1.1.7 // indirect
 )
 
+require (
+	github.com/docker/go-units v0.4.0 // indirect
+	github.com/go-task/slim-sprig v0.0.0-20210107165309-348f09dbbbc0 // indirect
+	github.com/godbus/dbus/v5 v5.0.3 // indirect
+	golang.org/x/mod v0.6.0-dev.0.20211013180041-c96bc1413d57 // indirect
+	golang.org/x/tools v0.1.8-0.20211029000441-d6a9af8af023 // indirect
+	golang.org/x/xerrors v0.0.0-20220517211312-f3a8303e98df // indirect
```
Comment (qdm12) [inline go.mod:193]: Please move these deps in the block above.

File: go.mod [hunk 10]
```diff
@@ -1,15 +1,15 @@
 module github.com/ChainSafe/gossamer
```
Comment (qdm12) [inline go.mod:1]: Can you open a PR with those dependency changes separately perhaps?
To keep a commit log making sense 😉

File: tests/stress/stress_test.go [hunk 11]
```diff
@@ -97,9 +97,10 @@ func stopNodes(cancel context.CancelFunc, runtimeErrors []<-chan error) {
 	}
 }
 
+// TODO: add tests against latest dev runtime
```
Comment (qdm12) [inline tests/stress/stress_test.go:100]: On that note, when should we do this? Is it the next upcoming PR?
Also should we keep older tests testing against the older runtime? 🤔

File: chain/dev/config.toml (hunk 1 above)
Comment (timwu20) [reply chain/dev/config.toml:21]: pretty sure? We usually specify the key when starting a node anyways.

File: cmd/gossamer/import_runtime.go (hunk 5 above)
Comment (timwu20) [reply cmd/gossamer/import_runtime.go:32]: this got changed in the new runtimes.  It would be nice if we marshaled this into a struct, and the json package would be able to interpret both capitalised and lowercase attribute names

File: cmd/gossamer/main_test.go (hunk 6 above)
Comment (timwu20) [reply cmd/gossamer/main_test.go:271]: Pretty sure it was wrong before.  The raw spec contains the code.

File: tests/stress/stress_test.go (hunk 11 above)
Comment (timwu20) [reply tests/stress/stress_test.go:100]: as mentioned in standup. The issues are created, and we should be tackling them as soon as we are finished with the current initiatives.

File: dot/network/gossip_test.go (hunk 7 above)
Comment (timwu20) [reply dot/network/gossip_test.go:89]: not sure. I'll take a look

File: cmd/gossamer/import_runtime.go (hunk 5 above)
Comment (qdm12) [reply cmd/gossamer/import_runtime.go:32]: yeah we can also have `UnmarshalJSON` methods and the json package would use those directly. Although how should we marshal them? Always with lowercase `system` or would that depend on the runtime version 😕 


@danforbes can you create an issue for this, to have all fields of the genesis as struct and have this runtime field with custom MarshalJSON/UnmarshalJSON?

File: dot/core/service_integration_test.go [hunk 12]
```diff
@@ -52,7 +51,7 @@ func balanceKey(t *testing.T, pub []byte) (bKey []byte) {
 
 func generateTestValidRemarkTxns(t *testing.T, pubKey []byte, accInfo types.AccountInfo) ([]byte, runtime.Instance) {
 	t.Helper()
-	projectRootPath := filepath.Join(utils.GetProjectRootPathTest(t), "chain/gssmr/genesis.json")
+	projectRootPath := utils.GetGssmrV3SubstrateGenesisRawPathTest(t)
```
Comment (jimjbrettj) [inline dot/core/service_integration_test.go:54]: Nice, first thing I looked for in potential conflicts of my branch was hard coded paths

File: dot/core/service_integration_test.go [hunk 13]
```diff
@@ -409,6 +408,7 @@ func TestHandleChainReorg_WithReorg_Transactions(t *testing.T) {
 	require.Equal(t, transaction.NewValidTransaction(tx, validity), pending[0])
 }
 
+// TODO: add test against latest gssmr runtime
```
Comment (jimjbrettj) [inline dot/core/service_integration_test.go:411]: Both these todos I am addressing

File: dot/sync/chain_processor_integration_test.go [hunk 14]
```diff
@@ -20,6 +20,7 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// TODO: add test against latest gssmr runtime
```
Comment (jimjbrettj) [inline dot/sync/chain_processor_integration_test.go:23]: Do we have issues created for these new tests? I am doing a few in core/services but not the whole codebase

File: lib/babe/build_integration_test.go [hunk 15]
```diff
@@ -100,6 +100,7 @@ func createTestBlock(t *testing.T, babeService *Service, parent *types.Header,
 	return block
 }
 
+// TODO: add test against latest dev runtime
```
Comment (jimjbrettj) [inline lib/babe/build_integration_test.go:103]: Maybe epic where we have issues for writing these test on a per-package basis? Like how we did the unit test refactor

File: tests/stress/stress_test.go (hunk 11 above)
Comment (jimjbrettj) [reply tests/stress/stress_test.go:100]: okay just saw this, addressed my above questions

**Review by jimjbrettj — APPROVED** [review]
Awesome work, really good step towards supporting latest runtime

File: dot/network/gossip_test.go (hunk 7 above)
Comment (timwu20) [reply dot/network/gossip_test.go]: this test doesn't work.  I must've just left that line in trying to debug it.

File: lib/babe/build_integration_test.go (hunk 15 above)
Comment (timwu20) [reply lib/babe/build_integration_test.go:103]: I created the issues.. I can group them into an epic if needed.

File: go.mod (hunk 10 above)
Comment (qdm12) [reply go.mod:1]: Probably also rebase on development since a bunch of deps were bumped on development (👀 centrifuge etc.)

**Review by qdm12 — COMMENTED** [review]
Just last comments:

1. let's merge the chaindb bump on development
2. let's rebase this on development
3. Any deps left to be bumped shall be bumped in a separate PR on development ideally

Also maybe rename the PR name to `add v0.9.20 runtime etc.` instead of using `embed` since I thought you were embedding the runtime in the binary 😄 That's probably just me, but a rework is cheap!

By the way, embedding genesis files in the binary could be fun in the future.

**Comment by timwu20** [comment]
> Just last comments:
> 
> 1. let's merge the chaindb bump on development
> 2. let's rebase this on development
> 3. Any deps left to be bumped shall be bumped in a separate PR on development ideally
> 
> Also maybe rename the PR name to `add v0.9.20 runtime etc.` instead of using `embed` since I thought you were embedding the runtime in the binary 😄 That's probably just me, but a rework is cheap!
> 
> By the way, embedding genesis files in the binary could be fun in the future.

I think we should move away from embedding runtimes in the future.

File: go.mod [hunk 16]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/chyeh/pubip v0.0.0-20170203095919-b7e679cf541c
 	github.com/cosmos/go-bip39 v1.0.0
 	github.com/dgraph-io/badger/v2 v2.2007.4
-	github.com/dgraph-io/ristretto v0.0.3-0.20200630154024-f66de99634de
+	github.com/dgraph-io/ristretto v0.1.0
```
Comment (qdm12) [inline go.mod:12]: Why do we need these dependency changes in this PR? 🤔

**Review by qdm12 — CHANGES_REQUESTED** [review]
Should we address the genesis `"system"` -> `"System"` to support both genesis formats before merging this?

**Comment by qdm12** [comment]
Also can you link the several new `// TODO` to the epic issue (or sub-issue) created?

**Comment by timwu20** [comment]
> Should we address the genesis `"system"` -> `"System"` to support both genesis formats before merging this?

We don't have genesis spec json files for kusama and polkadot chains that we have in the codebase, we use the raw ones.  This is only when we parse a genesis spec file, which we can enforce to be generated using the new attribute from substrate.

**Comment by timwu20** [comment]
> Also can you link the several new `// TODO` to the epic issue (or sub-issue) created?

The issues were linked before. Maybe the force pushing removed them from the conversation.

#2702 
#2703 
#2704 
#2705

File: go.mod (hunk 16 above)
Comment (timwu20) [reply go.mod]: resolved, not sure how that got in there

File: dot/core/service_integration_test.go (hunk 13 above)
Comment (qdm12) [inline dot/core/service_integration_test.go:411]:
```suggestion
// TODO: add test against latest gssmr runtime
// See https://github.com/ChainSafe/gossamer/issues/2702
```

File: dot/core/service_integration_test.go [hunk 17]
```diff
@@ -465,6 +465,7 @@ func TestMaintainTransactionPool_EmptyBlock(t *testing.T) {
 	require.Nil(t, head)
 }
 
+// TODO: add test against latest gssmr runtime
```
Comment (qdm12) [inline dot/core/service_integration_test.go:468]:
```suggestion
// TODO: add test against latest gssmr runtime
// See https://github.com/ChainSafe/gossamer/issues/2702
```

File: dot/sync/chain_processor_integration_test.go (hunk 14 above)
Comment (qdm12) [inline dot/sync/chain_processor_integration_test.go:23]:
```suggestion
// TODO: add test against latest gssmr runtime
// See https://github.com/ChainSafe/gossamer/issues/2703
```

File: dot/sync/chain_processor_integration_test.go [hunk 18]
```diff
@@ -79,6 +80,7 @@ func TestChainProcessor_HandleBlockResponse_ValidChain(t *testing.T) {
 	}
 }
 
+// TODO: add test against latest gssmr runtime
```
Comment (qdm12) [inline dot/sync/chain_processor_integration_test.go:83]:
```suggestion
// TODO: add test against latest gssmr runtime
// See https://github.com/ChainSafe/gossamer/issues/2703
```

File: dot/sync/chain_processor_integration_test.go [hunk 19]
```diff
@@ -154,6 +156,7 @@ func TestChainProcessor_HandleBlockResponse_NoBlockData(t *testing.T) {
 	require.Equal(t, ErrNilBlockData, err)
 }
 
+// TODO: add test against latest gssmr runtime
```
Comment (qdm12) [inline dot/sync/chain_processor_integration_test.go:159]:
```suggestion
// TODO: add test against latest gssmr runtime
// See https://github.com/ChainSafe/gossamer/issues/2703
```

File: dot/sync/chain_processor_integration_test.go [hunk 20]
```diff
@@ -183,6 +186,7 @@ func TestChainProcessor_HandleBlockResponse_BlockData(t *testing.T) {
 	}
 }
 
+// TODO: add test against latest gssmr runtime
```
Comment (qdm12) [inline dot/sync/chain_processor_integration_test.go:189]:
```suggestion
// TODO: add test against latest gssmr runtime
// See https://github.com/ChainSafe/gossamer/issues/2703
```

File: lib/babe/build_integration_test.go (hunk 15 above)
Comment (qdm12) [inline lib/babe/build_integration_test.go:103]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2704
```

File: lib/babe/build_integration_test.go [hunk 21]
```diff
@@ -139,6 +140,7 @@ func TestBuildBlock_ok(t *testing.T) {
 	require.Equal(t, 1, len(extsBytes))
 }
 
+// TODO: add test against latest dev runtime
```
Comment (qdm12) [inline lib/babe/build_integration_test.go:143]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2704
```

File: lib/babe/build_integration_test.go [hunk 22]
```diff
@@ -233,6 +235,7 @@ func TestApplyExtrinsic(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// TODO: add test against latest dev runtime
```
Comment (qdm12) [inline lib/babe/build_integration_test.go:238]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2704
```

File: lib/babe/build_integration_test.go [hunk 23]
```diff
@@ -406,6 +409,7 @@ func TestDecodeExtrinsicBody(t *testing.T) {
 	require.True(t, contains)
 }
 
+// TODO: add test against latest dev runtime
```
Comment (qdm12) [inline lib/babe/build_integration_test.go:412]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2704
```

File: lib/babe/verify_integration_test.go [hunk 24]
```diff
@@ -63,6 +63,7 @@ func newTestVerificationManager(t *testing.T, genCfg *types.BabeConfiguration) *
 	return vm
 }
 
+// TODO: add test against latest dev runtime
```
Comment (qdm12) [inline lib/babe/verify_integration_test.go:66]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2704
```

File: lib/babe/verify_integration_test.go [hunk 25]
```diff
@@ -74,6 +75,7 @@ func TestVerificationManager_OnDisabled_InvalidIndex(t *testing.T) {
 	require.Equal(t, err, ErrInvalidBlockProducerIndex)
 }
 
+// TODO: add test against latest dev runtime
```
Comment (qdm12) [inline lib/babe/verify_integration_test.go:78]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2704
```

File: lib/babe/verify_integration_test.go [hunk 26]
```diff
@@ -111,6 +113,7 @@ func TestVerificationManager_OnDisabled_NewDigest(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// TODO: add test against latest dev runtime
```
Comment (qdm12) [inline lib/babe/verify_integration_test.go:116]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2704
```

File: lib/babe/verify_integration_test.go [hunk 27]
```diff
@@ -168,6 +172,7 @@ func TestVerificationManager_VerifyBlock_Ok(t *testing.T) {
...
```
Comment (qdm12) [inline lib/babe/verify_integration_test.go:175]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2704
```

File: lib/babe/verify_integration_test.go [hunk 28]
```diff
@@ -276,6 +281,7 @@ func TestVerificationManager_VerifyBlock_MultipleEpochs(t *testing.T) {
...
```
Comment (qdm12) [inline lib/babe/verify_integration_test.go:284]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2704
```

File: lib/babe/verify_integration_test.go [hunk 29]
```diff
@@ -367,6 +374,7 @@ func TestVerifyPimarySlotWinner(t *testing.T) {
 	require.True(t, ok)
 }
 
+// TODO: add test against latest dev runtime
```
Comment (qdm12) [inline lib/babe/verify_integration_test.go:377]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2704
```

File: lib/babe/verify_integration_test.go [hunk 30]
```diff
@@ -386,6 +394,7 @@ func TestVerifyAuthorshipRight(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// TODO: add test against latest dev runtime
```
Comment (qdm12) [inline lib/babe/verify_integration_test.go:397]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2704
```

File: tests/polkadotjs_test/start_polkadotjs_test.go [hunk 31]
```diff
@@ -19,6 +19,7 @@ import (
 
 var polkadotSuite = "polkadot"
 
+// TODO: add test against latest dev runtime
```
Comment (qdm12) [inline tests/polkadotjs_test/start_polkadotjs_test.go:22]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2705
```

File: tests/rpc/rpc_02-author_test.go [hunk 32]
```diff
@@ -22,8 +22,9 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// TODO: add test against latest dev runtime
```
Comment (qdm12) [inline tests/rpc/rpc_02-author_test.go:25]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2705
```

File: tests/rpc/rpc_05-state_test.go [hunk 33]
```diff
@@ -161,8 +161,9 @@ func TestStateRPCResponseValidation(t *testing.T) { //nolint:tparallel
 	})
 }
 
+// TODO: add test against latest gssmr runtime
```
Comment (qdm12) [inline tests/rpc/rpc_05-state_test.go:164]:
```suggestion
// TODO: add test against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2705
```

File: tests/stress/stress_test.go [hunk 34]
```diff
@@ -238,12 +239,13 @@ func TestSync_SingleSyncingNode(t *testing.T) {
...
```
Comment (qdm12) [inline tests/stress/stress_test.go:242]:
```suggestion
// TODO: add tests against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2705
```

File: tests/stress/stress_test.go (hunk 11 above)
Comment (qdm12) [inline tests/stress/stress_test.go:100]:
```suggestion
// TODO: add tests against latest dev runtime
// See https://github.com/ChainSafe/gossamer/issues/2705
```

**Review by qdm12 — COMMENTED** [review]
Just a bunch of commit suggestions to add the issue links to each `// TODO` comment.

@timwu20 also please update #2705 to add tests for the latest dev runtime **and gssmr runtime** for all end to end tests (not just rpc since stress + polkadotjs tests are affected too)

**Comment by qdm12** [comment]
> We don't have genesis spec json files for kusama and polkadot chains that we have in the codebase, we use the raw ones. This is only when we parse a genesis spec file, which we can enforce to be generated using the new attribute from substrate.

That makes sense. So should we bother at all with supporting the older `system` field name?

**Comment by timwu20** [comment]
> Also please update #2705 to add tests for the latest dev runtime **and gssmr runtime** for all end to end tests (not just rpc since stress + polkadotjs tests are affected too)

This isn't necessary, the rpc tests only run on the dev runtime since it's a single node block producing environment.

**Comment by timwu20** [comment]
> > We don't have genesis spec json files for kusama and polkadot chains that we have in the codebase, we use the raw ones. This is only when we parse a genesis spec file, which we can enforce to be generated using the new attribute from substrate.
> 
> That makes sense. So should we bother at all with supporting the older `system` field name?

I don't think it's necessary.

**Comment by qdm12** [comment]
> This isn't necessary, the rpc tests only run on the dev runtime since it's a single node block producing environment.

I encountered [this TODO comment](https://github.com/ChainSafe/gossamer/pull/2543/files#diff-01e69913c9769890c2a8e2b1be16976db17acbd00ce2478dae23b4e7ef440fc4R164) mentioning `gssmr` in rpc tests, maybe it's a bad `// TODO`??

at least we need the issue updated to encompass rpc tests + stress tests + polkadotjs tests

**Comment by timwu20** [comment]
> > This isn't necessary, the rpc tests only run on the dev runtime since it's a single node block producing environment.
> 
> I encountered [this TODO comment](https://github.com/ChainSafe/gossamer/pull/2543/files#diff-01e69913c9769890c2a8e2b1be16976db17acbd00ce2478dae23b4e7ef440fc4R164) mentioning `gssmr` in rpc tests, maybe it's a bad `// TODO`?? This [suggestion](https://github.com/ChainSafe/gossamer/pull/2543#discussion_r944518034) changes it from gssmr to dev.
> 
> at least we need the issue updated to encompass rpc tests + stress tests + polkadotjs tests

Some of the tests continued to work when we updated the `gssmr` and `dev` chains in the tests. The issues I created were the ones that didn't work anymore, and we'll have to make code changes and/or revisit the tests in the upcoming sprint.

File: tests/rpc/rpc_05-state_test.go (hunk 33 above)
Comment (timwu20) [reply tests/rpc/rpc_05-state_test.go:164]: This is still `gssmr`, since we import the `gssmr` runtime in this tests.  We can look into the reason why we do this for this tests when we revisit in #2705

**Review by qdm12 — APPROVED** [review]
